core/lib/multivm/src/versions/vm_latest/bootloader/utils.rs

```diff
@@ -133,7 +157,147 @@ fn apply_l2_block_inner(
+        U256::from(interop_root.chain_id),
+        U256::from(interop_root.block_number),
+        U256::from(interop_root.sides.len()),
+    ];
+
+    u256_words.extend(interop_root.sides);
+    // println!("u256_words: {:?}", u256_words);
+    // println!(
+    //     "zipped 0 {:?}",
+    //     (interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE
+    //         ..interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE + u256_words.len())
+    //         .zip(u256_words.clone())
+    //         .collect::<Vec<_>>()[0]
+    // );
+    // println!(
+    //     "zipped 1 {:?}",
+    //     (interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE
+    //         ..interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE + u256_words.len())
+    //         .zip(u256_words.clone())
+    //         .collect::<Vec<_>>()[1]
+    // );
+    // println!(
+    //     "zipped 2 {:?}",
+    //     (interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE
+    //         ..interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE + u256_words.len())
+    //         .zip(u256_words.clone())
+    //         .collect::<Vec<_>>()[2]
+    // );
+    // println!(
+    //     "zipped 3 {:?}",
+    //     (interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE
+    //         ..interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE + u256_words.len())
+    //         .zip(u256_words.clone())
+    //         .collect::<Vec<_>>()[3]
+    // ); // Map the U256 words into memory slots
+    memory.extend(
+        (interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE
+            ..interop_root_slot + interop_root_offset * INTEROP_ROOT_SLOTS_SIZE + u256_words.len())
+            .zip(u256_words),
+    )
+}
+
+pub(crate) fn apply_interop_root_number_in_block_number(
+    memory: &mut BootloaderMemory,
+    subversion: MultiVmSubversion,
+    number_of_interop_roots: usize,
+    preexisting_blocks_number: usize,
+) {
+    let mut number_of_written_blocks = 0;
+    if let Some(_index) = memory.iter().position(|(slot, value)| {
+        if *slot < get_interop_blocks_begin_offset(subversion)
+            || *slot >= get_interop_root_offset(subversion)
+        {
+            return false;
+        }
+        println!("finding emptyslot: {}", slot);
+        println!("finding empty value: {}", value);
+        true
+        // *value != U256::from(1)
+    }) {
+        number_of_written_blocks += 1;
+    }
+    let mut first_empty_slot =
+        get_interop_blocks_begin_offset(subversion) + number_of_written_blocks;
+    if number_of_written_blocks == 0 {
+        first_empty_slot = get_interop_blocks_begin_offset(subversion) + preexisting_blocks_number;
+    }
+    println!("first_empty_slot: {}", first_empty_slot);
+    println!("number_of_interop_roots: {}", number_of_interop_roots);
+    let number_of_interop_roots_plus_one: U256 = (number_of_interop_roots + 1).into();
```

Comment:
@0xValera we add 1 to differentiate between empty blocks and non-existant blocks.